src/main/java/use_case/login/LoginInteractor2.java

```diff
@@ -17,12 +20,16 @@ public void execute(LoginInputData2 loginInputData2) {
                 LoginClass loginClass = new LoginClass();
                 while (accessToken == null || accessToken.isEmpty()) {
                     Thread.sleep(500); // Poll every 500 milliseconds
-                    accessToken = loginClass.getAccess_token();
+                    accessToken = loginClass.getAccess_token(); // Get the token
                 }
 
+                // Update CurrentUser with the access token
+                currentUser.setAccessToken(accessToken);
+
+                // Notify the presenter that login was successful
                 loginPresenter2.prepareSuccessView();
             } catch (Exception ex) {
-                throw new RuntimeException(ex);
+                throw new RuntimeException("Error during login: " + ex.getMessage(), ex);
             }
         }
     }
```

Comment:
Good thinking to add an error message in order to indicate where the error occurs